### DISCUSSION TITLE
Web & linux support in maui

### DISCUSSION THREAD

**Discussion by manishbisen** [dotnet/maui#1858] [Q&A]
When maui will support to web & linux platform ?

**Comment by GeraudFabien** [comment]
You may want to check:
https://github.com/dotnet/MobileBlazorBindings
https://platform.uno/

To answer:
 web never (you have blazor for that).
Linux maybe in .Net 7 maybe not. Depend if someone invest in it.

Related to:
https://github.com/dotnet/maui/discussions/1521
https://github.com/dotnet/maui/discussions/326
https://github.com/dotnet/maui/discussions/558
https://github.com/dotnet/maui/discussions/178
https://github.com/dotnet/maui/issues/1126

**Reply by manishbisen** [reply]
When Uno by c# can support to web then why maui can't ?

**Reply by manishbisen** [reply]
By blazor will it be kind of like mobile app or will run in mobile browser ?And will it be by single code base target to andriod, ios,mac,windows & web ?

**Reply by GeraudFabien** [reply]
Maui isn't for pixel perfect app. The goal of maui is to share the code not the ui.
Base on this goal why would they support web if you already can.
Also there is Maui blazor binding and uno if you want to share control with the missing target.

**Reply by GeraudFabien** [reply]
Blazor moblie binding run you're C# code "natively" but use webview to render HTML/CSS or run JS if you're blazor has JS. And yes it's the same UI everywhere since it's a web technology.

https://devblogs.microsoft.com/aspnet/hybrid-blazor-apps-in-mobile-blazor-bindings-july-update/
https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-4/#net-maui-blazor-apps
https://devblogs.microsoft.com/dotnet/announcing-net-maui-preview-4/#blazorwebview

**Reply by manishbisen** [reply]
So as per your comment do you mean web platform is already supported by maui ?

**Reply by GeraudFabien** [reply]
No but by blazor or WinUI. For maui there is not even plan to support it. But if you consider BlazorWebView and Uno Plateform you could share some control with web.

**Reply by osnipezzini** [reply]
We don't need for now a Pixel perfect for Linux, for me main use it's for configurators 

Uno it's a problem because don't have Shell then we need to design navigation for every new app

**Reply by osnipezzini** [reply]
Another problem is that Uno uses WinUI syntax, and their website documentation is horrible

**Reply by osnipezzini** [reply]
MobileBlazorBindings is basically a webapp, and I've had a problem since for example it's not possible to correctly access the host's file system
